src/runtime_src/driver/common/scheduler.h

```diff
@@ -33,7 +47,7 @@ namespace xrt_core { namespace scheduler {
  * Check sdaccel.ini for default overrides.
  */
 int
-init(xclDeviceHandle handle, const axlf* top);
+init(xclDeviceHandle handle, const axlf* top, bool encode);
```

Comment:
Why is there an encode argument here?   Is 'false' ever passed?

Reply:
In general this file is no longer exclusively about scheduler initialization.  I think I would have preferred leaving scheduler init alone and simply add another file and API for get_cus and get_dbg_ips_pair.   For sure the file name and namespace is now confusing.

Reply:
The xclLoadXclbin in zynq/user/shim.cpp pass 'false' to the scheduler init.
We could have another file which provides XCLBIN related APIs and namesapce.

Reply:
I still don't understand the encode argument.  Why will zynq pass 'false'?  How will it know that CUs are configured for dataflow?    Shouldn't it at least be passed the information and filter it out like x86 does?

Reply:
In 2019.1, we are not going to support dataflow for MPSoC. That is why I need to disable the encoding here. The zocl driver could not handle CU physical address with encode handshaking control in the lower unused bits.
We would remove the encode argument after we have dataflow support on MPSoC. Do you think this make sense?

Reply:
No. It still doesn't make sense.   The change should not have been at this level, but instead zocl should have adjusted the addresses in the same way xocl does before it actually uses them.   xocl simply checks the encoding, the clears it.  The checking is zocl is no-op since dataflow is not supported, but the clearing is not.   Is there maybe a reason this cannot be done in zocl?

Reply:
OK. I have made that change in zocl driver. I saw in the mb_scheduler.c. It has "addr = addr & ~(0xFF)" to clear encoding. I do the same in zocl for now.